packages/api/app/middleware.js

```diff
@@ -0,0 +1,26 @@
+const { EcsStartTaskError } = require('@cumulus/errors');
+const Logger = require('@cumulus/logger');
+
+const logger = new Logger({ sender: '@cumulus/api' });
+
+// Catch and send the error message down (instead of just 500: internal server error)
+// Need all 4 params, because that's how express knows this is the error handler
+// eslint-disable-next-line no-unused-vars
+const defaultErrorHandler = (error, req, res, next) => {
+  logger.error(error);
+  return res.boom.badRequest(error.message, error);
+};
+
+// eslint-disable-next-line no-unused-vars
+const asyncOperationEndpointErrorHandler = async (err, req, res, next) => {
```

Comment:
So as I was updating the error handling for the `/replays` endpoint, it occurred to me that we probably want to do the error handling consistently for all the endpoints which kick off async operations. I debated a few options but made the call that a custom error middleware for these endpoints might be best, because we can handle it in one place and just attach this middleware everywhere we define an endpoint for async operations. But interested in what others think

Reply:
Seems reasonable.